utils/graph.c

```diff
@@ -202,10 +205,11 @@ static int add_graph_event(struct uftrace_task_graph *tg, size_t node_size)
 }
 
 int graph_add_node(struct uftrace_task_graph *tg, int type, char *name,
-		   size_t node_size)
+		   size_t node_size,
+		   struct debug_location* loc)
 {
 	if (type == UFTRACE_ENTRY)
-		return add_graph_entry(tg, name, node_size);
+		return add_graph_entry(tg, name, node_size, loc);
 	else if (type == UFTRACE_EXIT)
```

Comment:
now I don't know what UFTRACE_EXIT means exactly. is it right it doesn't need loc?

Reply:
For graph processing, UFTRACE_ENTRY records add information about the function and UFTRACE_EXIT records update the execution time.  So I think it's ok to add the loc only to the ENTRY path.